cpp/src/arrow/python/deserialize.cc

```diff
@@ -256,13 +260,19 @@ Status DeserializeSet(PyObject* context, const Array& array, int64_t start_idx,
 Status ReadSerializedObject(io::RandomAccessFile* src, SerializedPyObject* out) {
   int64_t bytes_read;
   int32_t num_tensors;
+  int32_t num_ndarrays;
   int32_t num_buffers;
+
   // Read number of tensors
   RETURN_NOT_OK(
       src->Read(sizeof(int32_t), &bytes_read, reinterpret_cast<uint8_t*>(&num_tensors)));
+  RETURN_NOT_OK(
+      src->Read(sizeof(int32_t), &bytes_read, reinterpret_cast<uint8_t*>(&num_ndarrays)));
   RETURN_NOT_OK(
       src->Read(sizeof(int32_t), &bytes_read, reinterpret_cast<uint8_t*>(&num_buffers)));
 
+  // Align stream to 8-byte offset
+  RETURN_NOT_OK(ipc::AlignStream(src, ipc::kArrowIpcAlignment));
```

Comment:
I would like to look more closely at this before this patch is merged. I spent a bunch of time in https://github.com/apache/arrow/commit/c9ac8696a335f4e818c7356d9af77317150e94ce on this and so I want to make sure that stream alignment is happening at the highest level possible rather than leaking into lower-level implementation details

Reply:
@wesm Let me know about your thoughts on this, it seems that handling the alignment here is analogous to how the alignment for Tensors is handled down below in correspondence with https://github.com/apache/arrow/commit/c9ac8696a335f4e818c7356d9af77317150e94ce, so this looks good to me.

Reply:
I think this should probably be changed to take `InputStream* src` as input and handle alignment one level higher. I'm going to check out this branch and take a look

Reply:
Took a closer look and we can leave as is (the extra 4 bytes at the start of the object made the need for the alignment)